Add headless Amplify cli support

salihgueler opened the pull request.

salihgueler committed:
Initial try for hooks

salihgueler committed:
Initila, support for headless amplify

felangel committed:
feat: add hooks to integrate with amplify cli

salihgueler committed:
Merge branch 'add-headless-Amplify-CLI-support' into feat/headless-amplify-cli

salihgueler committed:
Merge pull request #2 from felangel/feat/headless-amplify-cli

feat: add hooks to integrate with amplify cli

salihgueler committed:
Add the api call.

salihgueler committed:
Add the tested version of the postgen

felangel reviewed (commented):
Thoughts on removing the .DS_Store? I think it can be gitignored

felangel commented on hooks/.dart_tool/package_config.json:
```diff
@@ -0,0 +1,140 @@
+{
```
The .dart_tool directory should be gitignored as well

felangel commented on hooks/.packages:
```diff
@@ -0,0 +1,28 @@
+# This file is deprecated. Tools should instead consume 
```
This file should be gitignored as well

felangel commented on hooks/post_gen.dart:
```diff
@@ -0,0 +1,176 @@
+  final flutterPubGetProgress = context.logger.progress(
+    'running "flutter pub get"',
+  );
+  final result = Process.runSync(
```
Thoughts on using Process.start for consistency?

salihgueler replied:
Makes sense 👍🏼

felangel commented on hooks/post_gen.dart:
```diff
@@ -0,0 +1,176 @@
+    'running "amplify init --yes"',
+  );
+
+  final amplifyResult = Process.runSync(
```
Same here

felangel commented on hooks/pubspec.yaml:
```diff
@@ -0,0 +1,7 @@
+  sdk: ">=2.12.0 <3.0.0"
+
+dependencies:
+  mason: ^0.1.0-dev.28
```
This can be bumped to the latest version as well

salihgueler committed:
Address some of the feedbacks.

salihgueler committed:
Update pubspec.yaml

salihgueler committed:
Update readmes.

salihgueler commented:
@felangel All comments are addressed!

felangel commented:
> @felangel All comments are addressed!

Looks like the `.DS_Store` is still there but otherwise LGTM!

felangel commented on .gitignore:
```diff
@@ -1 +1,3 @@
-.idea/
\ No newline at end of file
+.idea/
+hooks/.dart_tool
```
```suggestion
```

felangel commented on .gitignore:
```diff
@@ -1 +1,3 @@
\ No newline at end of file
+.idea/
+hooks/.dart_tool
+hooks/.packages
```
```suggestion
```

felangel replied:
I don’t think those are necessary since you added them to the hooks .gitignore

salihgueler replied:
> .DS_Store

That is odd, I don't see it in my system. But I added to the project's gitignore now.

salihgueler commented:
> .DS_Store

That is odd, I don't see it in my system. But I added to the project's gitignore now. EDIT: When I added the file to git ignore it showed itself in the VSCode but still not in the file system hahaha

salihgueler commented:
<img width="899" alt="Screenshot 2022-08-04 at 17 03 26" src="https://user-images.githubusercontent.com/24432752/182866715-68d97440-0e63-4c6f-87aa-a3245589e3f4.png">

salihgueler committed:
Add .DS_Store to gitignore

felangel commented:
Weird...it's still showing up in the diff.
Did you try `git rm --cached .DS_Store`?

<img width="581" alt="Screen Shot 2022-08-04 at 9 36 37 AM" src="https://user-images.githubusercontent.com/8855632/182874418-29fe4c92-d1d5-4635-8bf0-27bf2ed61247.png">

salihgueler committed:
REMOVE DS_Store

salihgueler commented:
> Weird...it's still showing up in the diff. Did you try `git rm --cached .DS_Store`?
> 
> <img alt="Screen Shot 2022-08-04 at 9 36 37 AM" width="581" src="https://user-images.githubusercontent.com/8855632/182874418-29fe4c92-d1d5-4635-8bf0-27bf2ed61247.png">

How about now? 😄

felangel commented:
> > Weird...it's still showing up in the diff. Did you try `git rm --cached .DS_Store`?
> > <img alt="Screen Shot 2022-08-04 at 9 36 37 AM" width="581" src="https://user-images.githubusercontent.com/8855632/182874418-29fe4c92-d1d5-4635-8bf0-27bf2ed61247.png">
> 
> How about now? 😄

All good now 🎉
